app.go

```diff
@@ -79,6 +79,14 @@ type App interface {
 	//
 	// Since: 2.3
 	SetCloudProvider(CloudProvider) // configure cloud for this app
+
+	// CustomMetadata allow you to get metadata specified at compile time
+	// It will contain the Release section if the binary was compiled in Release mode, the Development section otherwise
+	// The value can also be injected on the command line during compilation if you do not want to store them in your
+	// source repository
+	//
+	// Since: 2.3
+	CustomMetadata(name string) (custom string, exist bool)
```

Comment:
Would this not be better attached to the `AppMetadata` struct?

Reply:
AppMetadata is a structure not an interface. I think that hiding the data structure with an interface allow for more potential use case in the future with less chance of breaking API.

Reply:
I am not to sure how this would look like. AppMetadata is implemented in the fyne module while the metadata is currently provided by the app module. Attaching along the AppMetadata means that the implementation will be in fyne for custom metadata. It also means that the setter for it should be in fyne module or that we make the map public.

Reply:
Given that this is where the app metadata is stored it should at least be explored.
I don't see a strong reason not to follow the pattern that all other metadata has taken so far.